### PR TITLE
Add mobile registration

### PR DESCRIPTION
**Pull request by langleyd**
Add mobile registration and dispatch event for mobile postmessage dance

### CODE REVIEW

**Comment by pixlwave** [comment]
Any chance we could disable auto correction/capitalisation on the username field similar to what we did for MAS:

https://github.com/matrix-org/matrix-authentication-service/pull/312

File: src/components/structures/auth/Registration.tsx [hunk 1]
```diff
@@ -410,18 +417,33 @@ export default class Registration extends React.Component<IProps, IState> {
...
-                    identityServerUrl: this.state.matrixClient.getIdentityServerUrl(),
-                    accessToken,
-                },
-                this.state.formVals.password!,
-            );
+            if (this.props.mobileRegister) {
+                const mobileResponse: MobileRegistrationResponse = {
+                    user_id: userId,
+                    home_server: this.state.matrixClient.getHomeserverUrl(),
+                    access_token: accessToken,
```
Comment (bmarty) [inline src/components/structures/auth/Registration.tsx:424]: Can we have the refresh token as well?

File: src/components/structures/MatrixChat.tsx [hunk 2]
```diff
@@ -976,12 +981,15 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
             newState.register_client_secret = params.client_secret;
             newState.register_session_id = params.session_id;
             newState.register_id_sid = params.sid;
+            newState.register_id_sid = params.sid;
```
Comment (bmarty) [inline src/components/structures/MatrixChat.tsx:984]: Looks like a duplicated line

Comment (langleyd) [reply src/components/structures/MatrixChat.tsx:984]: Thanks

**Review by dbkr — APPROVED** [review]
Looks plausible

File: src/components/structures/auth/Registration.tsx (hunk 1 above)
Comment (langleyd) [reply src/components/structures/auth/Registration.tsx:424]: discussed off, not needed for now.

**Pull request merged**